crates/sui-adapter/src/adapter.rs

```diff
@@ -1345,15 +1365,36 @@ fn missing_unwrapped_msg(id: &ObjectID) -> String {
     )
 }
 
-fn convert_type_argument_error(idx: usize, error: VMError) -> ExecutionError {
+fn convert_type_argument_error(
+    idx: usize,
+    error: VMError,
+    vm_resolver: impl Fn(&ModuleId, FunctionDefinitionIndex) -> Option<Identifier>,
+) -> ExecutionError {
     use move_core_types::vm_status::StatusCode;
     use sui_types::messages::EntryTypeArgumentErrorKind;
     let kind = match error.major_status() {
         StatusCode::LINKER_ERROR => EntryTypeArgumentErrorKind::ModuleNotFound,
         StatusCode::TYPE_RESOLUTION_FAILURE => EntryTypeArgumentErrorKind::TypeNotFound,
         StatusCode::NUMBER_OF_TYPE_ARGUMENTS_MISMATCH => EntryTypeArgumentErrorKind::ArityMismatch,
         StatusCode::CONSTRAINT_NOT_SATISFIED => EntryTypeArgumentErrorKind::ConstraintNotSatisfied,
-        _ => return error.into(),
+        _ => return convert_vm_error(error, vm_resolver),
     };
     ExecutionErrorKind::entry_type_argument_error(idx as TypeParameterIndex, kind).into()
 }
+
+pub fn with_vm_resolver<
+    'r,
+    E: Debug,
+    S: ResourceResolver<Error = E> + ModuleResolver<Error = E>,
+>(
+    vm: &'r MoveVM,
+    state_view: &'r S,
+) -> impl Fn(&ModuleId, FunctionDefinitionIndex) -> Option<Identifier> + 'r {
+    |m: &ModuleId, idx: FunctionDefinitionIndex| {
+        let module = vm.load_module(m, state_view).ok()?;
```

Comment:
I am wondering now... is this safe? I'm not sure that it is. I don't think we have a test for it, but could you try publishing a module that has a Move level verifier error?

There won't be a module to load in that case. I am logging off here shortly, but can help craft such an example tomorrow if you need the help :)

Reply:
Not sure I follow what could be unsafe here, are you referring to this line and the call to `ok()`?

```rust
let module = vm.load_module(m, state_view).ok()?;
```

Because I believe `ok()` just [converts `Result` to `Option`](https://sourcegraph.com/github.com/rust-lang/rust/-/blob/library/core/src/result.rs?L642-652&subtree=true) (don't think this is unsafe?)

I can look into a test, but it'd be great to understand what looks unsafe here.

Reply:
>  it'd be great to understand what looks unsafe here.

Unsafe here meaning it won't do what you want :) not like, panic or anything. 

Let's say you publish this module 
```
// example.mvir
module 0x0.m {

fun no(): u64 {
  0u8
}

}
```
This will go through and eventually in the adapters `verify_and_link` it will call through the VM to `session.publish_module_bundle` which eventually calls the bytecode verifier. That will fail, and produce a `VMError` which points to `0x0::m::no` at offset `0`. We will then try to convert the error, and then load the module. But that module won't exist! So instead of getting a function name of `Some("no")` you will get `None`

Reply:
Got it, so we are possibly missing a case where we could report module and function but don’t (false negative) for that code path. 

Proposal though: we weren’t doing this before anyway right (reporting functions for this case you're describing), so this isn’t making the state of things worse, and with these changes we do fix up a good number of MoveAbort issues. How about I file an issue for this case, assign it, and investigate, and handle it properly separately after this bigger change? I also want to bring back module/function location reporting for the id_leak_verifier too.

Reply:
> How about I file an issue for this case, assign it, and investigate, and handle it properly separately after this bigger change?

Sure, I can take a look after this lands :)